File: extruturadedados/REVISÃO 3/REVISÃO/Decisão/Palpite_Jogo_do_bicho.py
import random

bichos = [
    "Avestruz", "Águia", "Burro", "Borboleta", "Cachorro",
    "Cabra", "Carneiro", "Camelo", "Cobra", "Coelho",
    "Cavalo", "Elefante", "Galo", "Gato", "Jacaré",
    "Leão", "Macaco", "Porco", "Pavão", "Peru",
    "Touro", "Tigre", "Urso", "Veado", "Vaca"
]

def gerar_palpite():
    numero = random.randint(0, 9999)  
    dois_ultimos = numero % 100
    grupo = (dois_ultimos // 4) % 25
    bicho = bichos[grupo]
    
    print(f"Número: {numero:04d}")
    print(f"Bicho: {bicho}")
    print(f"Grupo: {grupo + 1}")

def analisar_numero():
    try:
        numero = int(input("Digite um número (0000-9999): "))
        if 0 <= numero <= 9999:
            dois_ultimos = numero % 100
            grupo = (dois_ultimos // 4) % 25
            bicho = bichos[grupo]
            
            print(f"Número: {numero:04d}")
            print(f"Bicho: {bicho}")
            print(f"Grupo: {grupo + 1}")
        else:
            print("Número inválido! Digite entre 0000 e 9999.")
    except ValueError:
        print("Número inválido!")

def main():
    print("GERADOR DE PALPITES - JOGO DO BICHO")
    
    while True:
        print("\n1 - Gerar palpite")
        print("2 - Analisar número")
        print("3 - Sair")
        opcao = input("Escolha: ")
        
        if opcao == "1":
            gerar_palpite()
        elif opcao == "2":
            analisar_numero()
        elif opcao == "3":
            print("Até mais!")
            break
        else:
            print("Opção inválida!")

if __name__ == "__main__":
    main()

File: extruturadedados/REVISÃO 3/REVISÃO/Decisão/Palpite_Jogo_do_bicho.java
import java.util.Random;
import java.util.Scanner;

public class Palpite_Jogo_do_bicho {
    
    public static void main(String[] args) {
        String[] bichos = {
            "Avestruz", "Águia", "Burro", "Borboleta", "Cachorro",
            "Cabra", "Carneiro", "Camelo", "Cobra", "Coelho",
            "Cavalo", "Elefante", "Galo", "Gato", "Jacaré",
            "Leão", "Macaco", "Porco", "Pavão", "Peru",
            "Touro", "Tigre", "Urso", "Veado", "Vaca"
        };

        System.out.println("GERADOR DE PALPITES - JOGO DO BICHO");
        
        try (Scanner scanner = new Scanner(System.in)) {
            OUTER:
            while (true) {
                System.out.println("\n1 - Gerar palpite");
                System.out.println("2 - Analisar número");
                System.out.println("3 - Sair");
                System.out.print("Escolha: ");
                String opcao = scanner.nextLine();
                switch (opcao) {
                    case "1" -> gerarPalpite(bichos);
                    case "2" -> analisarNumero(bichos, scanner);
                    case "3" -> {
                        System.out.println("Até mais!");
                        break OUTER;
                    }
                    default -> System.out.println("Opção inválida!");
                }
            }
        }
    }
    
    static void gerarPalpite(String[] bichos) {
        Random random = new Random();
        int numero = random.nextInt(10000);
        
        int doisUltimos = numero % 100;
        int grupo = (doisUltimos / 4) % 25;
        String bicho = bichos[grupo];
        
        System.out.printf("Número: %04d\n", numero);
        System.out.println("Bicho: " + bicho);
        System.out.println("Grupo: " + (grupo + 1));
    }
    
    static void analisarNumero(String[] bichos, Scanner scanner) {
        System.out.print("Digite um número (0000-9999): ");
        String input = scanner.nextLine();
        
        try {
            int numero = Integer.parseInt(input);
            if (numero >= 0 && numero <= 9999) {
                int doisUltimos = numero % 100;
                int grupo = (doisUltimos / 4) % 25;
                String bicho = bichos[grupo];
                
                System.out.printf("Número: %04d\n", numero);
                System.out.println("Bicho: " + bicho);
                System.out.println("Grupo: " + (grupo + 1));
            } else {
                System.out.println("Número inválido! Digite entre 0000 e 9999.");
            }
        } catch (NumberFormatException e) {
            System.out.println("Número inválido!");
        }
    }
}